src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -417,6 +418,107 @@ class C
         }
 
         [Fact]
+        [WorkItem(217740, "https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=217740")]
+        public void DateTimeConstantAttribute()
+        {
+            #region "Source"
+            var source = @"
+using System;
+using System.Runtime.CompilerServices;
+
+public class Bar
+{
+    public void Method([DateTimeConstant(-1)]DateTime p1) { }
```

Comment:
I already manually verified the behavior when referencing metadata. The behavior is consistent, an error is reported (C# used to report `error CS1501: No overload for method 'Method' takes 0 arguments` and VB used to report `error BC30455: Argument not specified for parameter 'p' of 'Public Function Method(p As Date = #12:00:00 AM#) As Date'`).
I cannot do compilation reference with native compiler. 

I added a test for both scenarios (C# and VB).
I'll add test for "double attribute" concern.